### ISSUE TITLE
aws_lambda_function data source never updates, especially when qualifier changes

### ISSUE THREAD

**Issue by t-pascal** [terraform-providers/terraform-provider-aws#4593]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
$ terraform -v
Terraform v0.11.7
+ provider.aws v1.19.0
```

### Affected Resource(s)

* aws_lambda_function (data source)

### Terraform Configuration Files

```hcl
data "aws_lambda_function" "mylambda" {
  function_name = "my_function"
  qualifier     = "1"
}

output "mylambda_version" {
  value = "${data.aws_lambda_function.mylambda.qualified_arn}"
}
```

### Expected Behavior

Apply:
mylambda_version  = arn:aws:lambda:us-east-1:123456789:function:my_function:1

Change value to 2 and apply:
mylambda_version  = arn:aws:lambda:us-east-1:123456789:function:my_function:2

### Actual Behavior

Apply:

```
$terraform apply
provider.aws.region
  The region where AWS operations will take place. Examples
  are us-east-1, us-west-2, etc.

  Default: us-east-1
  Enter a value:

data.aws_lambda_function.mylambda: Refreshing state...

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

mylambda_version = arn:aws:lambda:us-east-1:12345678:function:my_function:1
```

Change qualifier:

```hcl
data "aws_lambda_function" "mylambda" {
  function_name = "my_function"
  qualifier     = "2"
}

output "mylambda_version" {
  value = "${data.aws_lambda_function.mylambda.qualified_arn}"
}
```

Reapply:

```
$terraform apply
provider.aws.region
  The region where AWS operations will take place. Examples
  are us-east-1, us-west-2, etc.

  Default: us-east-1
  Enter a value:

data.aws_lambda_function.mylambda: Refreshing state...

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

mylambda_version = arn:aws:lambda:us-east-1:123456789:function:my_function:1
```

Taint???

```
terraform taint data.aws_lambda_function.mylambda
Resource 'data.aws_lambda_function.mylambda' cannot be tainted
```

### Steps to Reproduce

1. `terraform apply`
2. Verify outputs
3. Change qualifier
4. `terraform apply`
5. Verify outputs are unchanged

6. In fact, data resource seems to never update again

### References

See #2984

**Comment by t-pascal** [comment]
I think I have verified that this bug in the data resource only returns the $LATEST version of the lambda, apparently ignoring the QUALIFIER flag.

**Comment by t-pascal** [comment]
Confirmed that this was fixed in 1.21.0 (2018 31 May) in PR #4654
